temporal/tracing.go

```diff
@@ -1,11 +1,136 @@
 package temporal
 
 import (
+	"context"
+	"github.com/samber/lo"
 	temporalotel "go.temporal.io/sdk/contrib/opentelemetry"
+	"go.temporal.io/sdk/converter"
 	"go.temporal.io/sdk/interceptor"
+	"go.temporal.io/sdk/workflow"
+	"strconv"
 )
 
+type (
+	workflowTraceParametersKey        struct{}
```

Comment:
This file was meant for tracing (as in the Open Telemetry project), not for general metrics - I would not group them together 😄

Reply:
my bad. i've extracted all that functionality into observability file